core/storage/trie/polkadot_trie/polkadot_trie.hpp

```diff
@@ -73,6 +74,14 @@ namespace kagome::storage::trie {
     std::unique_ptr<BufferMapCursor> cursor() final {
       return trieCursor();
     }
+
+    inline static outcome::result<void> defaultNodeRetrieveFunctor(
+        NodePtr &parent) {
+      if (parent && parent->isDummy()) {
+        BOOST_ASSERT(!"Dummy node unexpected.");
+      }
```

Comment:
In release it will came empty if-statement. 
```suggestion
      BOOST_ASSERT(parent == nullptr or parent->isDummy());
```